pkg/controller/monitor/cnr_indicator.go

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2022 The Katalyst Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitor
+
+import (
+	"time"
+
+	"k8s.io/klog/v2"
+
+	"github.com/kubewharf/katalyst-api/pkg/apis/node/v1alpha1"
+	"github.com/kubewharf/katalyst-core/pkg/config/generic"
+	"github.com/kubewharf/katalyst-core/pkg/metrics"
+	"github.com/kubewharf/katalyst-core/pkg/util/native"
+	"github.com/kubewharf/katalyst-core/pkg/util/qos"
+)
+
+const (
+	metricsNameCNRReportAnomaly  = "cnr_report_anomaly"
+	metricsNameCNRReportLantency = "cnr_report_latency"
+)
+
+const (
+	// reasonNumaExclusiveAnomaly is the reason for numa exclusive anomaly
+	// when numa_binding and numa_exclusive are both set
+	// the pod with numa_binding and numa_exclusive shares the numa with other pods
+	reasonNumaExclusiveAnomaly = "The pod with numa_binding and numa_exclusive shares the numa with other pods"
+	// reasonNumaAllocatableSumAnomaly is the reason for numa allocatable sum anomaly
+	// when the node's sum of numa allocatable is not equal to the node allocatable
+	reasonNumaAllocatableSumAnomaly = "The node's sum of numa allocatable is not equal to the node allocatable"
```

Comment:
Simplify the reason to a short string like "NumaAllocatableSumAnomaly" since long strings can be challenging to filter in Grafana or Prometheus.